### ISSUE TITLE
[Bug] Incorrect relativePath for 404 pages

### ISSUE THREAD

**Issue by BirkhoffLee** [Snapstromegon/eleventy-plugin-rollup#11]
`/404.html` are used across all 404 pages, including `https://example.com/posts/hello-world`, this plugin resolves the target JS URL by relative paths, so scripts on 404 pages will not work.

Potential solutions:

1. Extend `scriptGenerator`: pass `eleventyInstance` to the callback as well so the user can customise the final resolution for the JS for each different context. In this example:
    ```diff
    +++ node_modules/eleventy-plugin-rollup/index.js	2023-02-11 15:46:32
    @@ -180 +180 @@
    -    return this.scriptGenerator(relativePath);
    +    return this.scriptGenerator(relativePath, eleventyInstance); // backwards compatible
    ```
    ```js
    eleventyConfig.addPlugin(rollupPlugin, {
      rollupOptions: "rollup.config.js",
      scriptGenerator: (relativeUrl, eleventyInstance) => {
        if (eleventyInstance.page.url === '/404.html') {
          relativeUrl = "/" + relativeUrl;
        }
    
        return `<script async defer src="${relativeUrl}"></script>`
      }
    });
    ```
2. Hardcode 404 logic to use absolute URL in this plugin.
    ```js
    if (eleventyInstance.page.url === '/404.html') {
      relativePath = "/" + relativePath;
    }
    ```

**Comment by Snapstromegon** [comment]
Thanks for the bug report @BirkhoffLee.

I see what you mean and what your usecase is and I think this is also important in other usecases (like SPAs). I personally dislike your second potential solution, because it involves hardcoding just one case, which probably feel like "magic" to users.

Your first solution on the other hand is pretty good IMO and I would be glad to accept a PR for that.
In addition (as a separate contribution by me or anyone else), I'd propose to amend the current `scriptGenerator` in a way that it can resolve all scripts as absolute paths relative to your eleventy output dir. This should also fix the issue in a fairly elegant way. I'm currently looking into how to get that information.

**Comment by Snapstromegon** [comment]
@BirkhoffLee I just published version 1.3.0 and it contains your patches and the ability to use absolute script paths for imports natively.
